scipy/stats/_stats_py.py

```diff
@@ -1606,23 +1608,29 @@ def skewtest(a, axis=0, nan_policy='propagate', alternative='two-sided'):
     agree fairly closely, even for our small sample.
 
     """
+    xp = array_namespace(a)
+    a, axis = _chk_asarray(a, axis, xp=xp)
+
     b2 = skew(a, axis)
     n = a.shape[axis]
     if n < 8:
         raise ValueError(
-            "skewtest is not valid with less than 8 samples; %i samples"
-            " were given." % int(n))
+            f"skewtest is not valid with less than 8 samples; {n} samples were given.")
```

Comment:
Would you mind if we tweaked the wording while we're here?
```suggestion
            f"`skewtest` is not valid with fewer than eight observations; {n} observations were given.")
```
It will need a corresponding adjustment in `_axis_nan_policy.py` and its test file.

The important part is "observations" vs "samples". To avoid ambiguity, I try to use "sample" only to mean "a set of observations" and never as a synonym for "observation".